src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1249,9 +1387,111 @@ Containers::Optional<MaterialData> AssimpImporter::doMaterial(const UnsignedInt
+                    else if(property.mDataLength/4 == 2)
+                        type = MaterialAttributeType::Vector2i;
+                    else if(property.mDataLength/4 == 3)
+                        type = MaterialAttributeType::Vector3i;
+                    else if(property.mDataLength/4 == 4)
+                        type = MaterialAttributeType::Vector4i;
+                    else {
+                        Warning{} << "Trade::AssimpImporter::material(): property" << key << "is an integer array of" << property.mDataLength/4 << "items, saving as a typeless buffer";
+                        /* Using Pointer to indicate this is a buffer. String
+                           would indicate aiString which needs special handling
+                           for the length prefix. This still becomes an owned
+                           String later. */
+                        type = MaterialAttributeType::Pointer;
+                    }
+                } else if(property.mType == aiPTI_Float) {
+                    if(property.mDataLength/4 == 1)
+                        type = MaterialAttributeType::Float;
+                    else if(property.mDataLength/4 == 2)
+                        type = MaterialAttributeType::Vector2;
+                    else if(property.mDataLength/4 == 3)
+                        type = MaterialAttributeType::Vector3;
+                    else if(property.mDataLength/4 == 4)
+                        type = MaterialAttributeType::Vector4;
+                    else {
+                        Warning{} << "Trade::AssimpImporter::material(): property" << key << "is a float array of" << property.mDataLength/4 << "items, saving as a typeless buffer";
+                        /* See comment above */
+                        type = MaterialAttributeType::Pointer;
+                    }
+                } else if(property.mType == aiPTI_Double) {
+                    /** @todo This shouldn't happen without compiling Assimp
+                        with double support. But then the importer would only
+                        produce garbage because we assume ai_real equals float
+                        everywhere. */
+                    Warning{} << "Trade::AssimpImporter::material():" << key << "is a double precision property, saving as a typeless buffer";
+                    /* See comment above */
+                    type = MaterialAttributeType::Pointer;
+                } else if(property.mType == aiPTI_Buffer) {
+                    if(property.mDataLength == 1) {
+                        /* The glTF importer writes bool properties as buffers
+                           with size 1, when all the other importers write them
+                           as ints. Currently there is no importer that
+                           writes opaque buffer data, so we try to detect this.
+                           Funny library. */
+                        type = MaterialAttributeType::Bool;
```

Comment:
I was wondering if we should handle this at all. It seemed convenient for the test, but it kind of goes against the idea of "raw" material data if we're *already* special-casing Assimp bugs here.

Reply:
Good point, actually. This might totally change again in Assimp 5.1.2 and having special cases like this would only generate more work for us. Remove, then :)

Related: I'm thinking that adding a `MaterialAttributeType::Buffer` might make sense, for binary data that can't really be viewed as strings (like, printing those via `magnum-sceneconverter --info-materials` to the terminal could garble it up completely).

Reply:
Makes sense to have 👍 I removed the special-casing and added a todo about `MaterialAttributeType::Buffer`.